drivers/ipm/ipm_mcux.c

```diff
@@ -143,10 +148,34 @@ static int mcux_mailbox_ipm_set_enabled(const struct device *d, int enable)
 	return 0;
 }
 
+static inline int mcux_mailbox_reset(const struct device *dev)
```

Comment:
Should this be conditional on `MAILBOX_USES_RESET`?

Reply:
no, I think that having #ifdefs and things in drivers is harder to read and maintain, and if MAILBOX_USES_RESET is false I would imagine the compiler will realize this function will always just return 0 and optimize it accordingly

Reply:
As long as it doesn't result in a compiler warning 😉

Reply:
Why would it? Things like compiler warnings and errors is what I try to avoid by not using #ifdefs